http/responseemitter.go

```diff
@@ -136,7 +145,14 @@ func (re *responseEmitter) SetLength(l uint64) {
 
 func (re *responseEmitter) Close() error {
 	re.once.Do(func() { re.preamble(nil) })
-	// can't close HTTP connections
+
+	select {
+	case <-re.closeWait:
+		return nil // already closed
+	default:
+		close(re.closeWait)
```

Comment:
Just to check, is `Close` supposed to be threadsafe? As is, it's not (if it should be, we can just use a once).

Reply:
Not sure, but if in doubt, go threadsafe.